Entrega proyecto 1

Sly-Agustin opened the pull request:
### Entrega Proyecto 1

- Tema Elegido:  2
- Link a GitHub Pages: [link](https://sly-agustin.github.io/proyecto-1/) 

### Aclaraciones

<!-- de ser necesario -->

Sly-Agustin committed:
commit de prueba

commit de prueba para ver que esté configurado de manera correcta la aplicación de escritorio de github

Sly-Agustin committed:
Cambios en HTML y CSS

- Estructurado mejor el html
- Agregados estilos CSS para fuentes y botones
- Agregado el modo nocturno beta en JS

Sly-Agustin committed:
Cambios en CSS

- Agregado Bootstrap
- Refinado el modo nocturno
- Agregadas imágenes a las carpetas de imágenes para CSS

Sly-Agustin committed:
Cambios a html

- Agregadas algunas cosas a la estructura html
- Agregada lista de pendientes
- Revisado el código CSS

Sly-Agustin committed:
Cambios en HTML, CSS y JS

- Cambiados los botones que usaban imagen de background por colores
- Agregados hover a los botones
- Mejorado el modo nocturno
- Agregados estilos CSS
- Agregada tabla con puntuaciones por operación realizada

Sly-Agustin committed:
Modificaciones

- Cambios en la interfaz
- Ya no se usan tablas para los puntajes, ahora se usan cards
- Agregado navbar
- Agregadas imágenes

Sly-Agustin committed:
Cambios

- Mejorado el modo nocturno
- Agregados estilos css
- Agregadas funciones javascript (con y sin jquery)
- Header completado (con modo nocturno)
- Agreado footer (con modo nocturno)
- Cambios a la estructura html

Sly-Agustin committed:
Juego funcional (no entrega final)

- Agregada lógica en JS para ver si los resultados están bien.
- Agregado timer
- Agregadas tablas de prueba (actualmente no son interactivas)

Sly-Agustin committed:
Cambios a la implementación del cálculo

-  Cambiado el JS para que ya no funcione con una cantidad fixeada de operandos. Ahora utiliza un arreglo con todos los operandos dentro e itera por el mismo para poder determinar un resultado, antes usaba dos variables que solo nos permitía hacer una operación sobre dos numeros.

Sly-Agustin committed:
Separación de archivos y localStorage

- Se agregó localStorage para poder recordar si el usuario utilizaba modo nocturno
- Se agregó un archivo "darkMode.js" con todo lo relacionado al modo nocturno

Sly-Agustin committed:
Mejoras a la interfaz

- Mejorada la interfaz (luce mas limpia)

Sly-Agustin committed:
Cambios

Sly-Agustin committed:
Actualizado readme

Sly-Agustin committed:
Cambios

- Cambios al readme
- Cambios al html para que sea un html5 válido

Sly-Agustin committed:
Link del video

- Agregado el link del video al readme

MarianoTucat commented on js/app.js:
```diff
@@ -1 +1,279 @@
+                }
+            }
+        }
+    }
```
JS tiene `switch' ;)

MarianoTucat reviewed (changes requested):
# Proyecto 1 - Sly

1. HTML: Bien
2. CSS: Regular
3. Javascript: Bien
4. Manejo DOM: Bien
5. Visualmente Atractiva: Regular
6. Local Storage: Bien
7. PR a tiempo: Si
8. GitHub Pages: Si
9. Video: Bien
Comentarios: Problemas de CSS, el boton Jugar, incluso deshabilitado, sigue reaccionando al hover como si se pudiera apretar. No se detecta el fin de juego completamente, se puede seguir, jugando infinitamente, resolviendo y apretando enter. Lo mismo, se puede sumar puntos sin ingresar nada.
Resultado: Re-entrega

Sly-Agustin committed:
Fixes y cambios

- El juego ahora hace uso de un estado para determinar si se está jugando o no.
- Arreglado que se pudiera seguir jugando con el enter incluso cuando le juego había terminado (usando el estado anteriormente dicho).
- Mejoradas las divisiones, ahora siempre son enteras y nunca pueden dar 0 o 1 para que sean de mayor dificultad, también aumentado el rango de números.
- Cambiada la parte html que describe las divisiones para matchear con los cambios realizados.

Sly-Agustin committed:
Cambios a la implementacion

- Las imágenes de las operaciones ya no togglean clases CSS, ahora se modifica el source con javascript, de manera que además de ser un documento HTML5 válido también no saltan más errores en consola. Además no saltan errores de imagen en edge como antes.
- Pequeños cambios a CSS

Sly-Agustin committed:
Cambios en apariencia

- Cambios en html para hacerlo visualmente más agradable.
- Agregado un label que muestra si la respuesta fue correcta o no
- Agregado container general para contener al resto.
- Cambios y agregados en CSS
- Cambios en darkMode javascript

Sly-Agustin committed:
Cambios en timer

- Agregado CSS y javascript para que el timer se ponga de color rojo cuando quede poco tiempo
- Agregadas fuentes con sombra
- Agregado color a las respuestas correctas/incorrectas

Sly-Agustin committed:
Html

- Agregado modal de fin de juego

Sly-Agustin committed:
Update app.js

- Arreglado el timer que estaba mal puesto

Sly-Agustin committed:
Fix modal

- Solucionado que el modal mostrara con el color incorrecto en modo nocturno

Sly-Agustin committed:
Cambios a la jugabilidad

- Las operaciones ya no pueden dar como resultado 0 para evitar trampas manteniendo el enter.

Sly-Agustin committed:
Pequeño cambio a html

- Pequeño cambio a HTML porque por no era HTML5 compatible por un error de copy-paste.

Sly-Agustin commented:
No se si es necesario un PR nuevo o seguimos con este. Voy a asumir que seguimos con este y comentar lo que se agrego, si bien todo lo que fuí agregando o corrigiendo se puede ver en los commits esto sería un resumen:
- Se corrigió el problema del botón de CSS agregando la propiedad de que además el botón tiene que estar activo con ":enabled".
- El fin del juego se detecta correctamente ahora. Se transfirió la responsabilidad de iniciar y finalizar el juego a la clase "Juego" y en la misma se agregó un estado para identificar si está jugando o no, si se quiere ver más en detalle mirar el código pero en esencia ahora la clase "Juego" se encarga de todo lo relacionado al mismo (como debería haber sido en un principio).
- Se corrigió que se pudiera sumar puntos sin ingresar nada. Esto más que un fix fue un cambio, cuando uno no ingresaba nada se tomaba como si fuera un 0 y algunas operaciones podían dar 0 de manera que era más fácil sumar puntos apretando enter que resolviendo las fórmulas. Ahora ninguna operación puede dar como resultado 0, de todas formas era bastante trivial darle a un usuario resolver, por ejemplo, 8-8.

Cosas adicionales:
- Cambié como se mostraban las imágenes de las operaciones, antes eran una imagen sin source y con un background content para mostrar la imagen, esto causaba problemas en mozilla y edge ya que los mismos no permiten imágenes en content. Ahora las imágenes que deben cambiar con el modo nocturno se muestran según un source y se cambia el mismo con javascript en caso de cambiar de modo. Sigue siendo HTML5 válido con la diferencia que ya no tenemos errores de consola. Para tener una referencia antes se cambiaba togleando clases CSS con javascript.
- Las divisiones ahora siempre son enteras.
- Se cambio un poco la estructura HTML para hacerla más agradable.
- El timer ahora se muestra en rojo cuando quede poco tiempo.
- Se agregó un pequeño panel que nos indica si respondimos correctamente o no.
- Se agregó un modal que salta al final del juego con la puntuación que hicimos.
- Se cambió el banner de facebook, ahora en lugar de compartir la página de los desarrolladores nos comparte la página del juego.

Sly-Agustin committed:
Barra de progreso

- Agregada una barra dinámica que decrece a medida que pasa el tiempo
- Borrado código innecesario

Sly-Agustin committed:
Clean code y modificación a la jugabilidad

- Pasada de cleancode por archivos javascript
- Limpiados ciertos comentarios
- Cambiados ciertos aspectos de la jugabilidad, ya no hay multiplicaciones, sumas ni restas con 1 (mínimo 2 ahora).

MarianoTucat reviewed (approved):
# Reentrega Proyecto 1 - Sly

1. HTML: Bien
2. CSS: Bien
3. Javascript: Bien
4. Manejo DOM: Bien
5. Visualmente Atractiva: Bien
6. Local Storage: Bien
7. PR a tiempo: Si
8. GitHub Pages: Si
9. Video: Bien
Comentarios: Correcto
Estrellas: 🌟🌟
Resultado: Aprobado
